core/trino-main/src/test/java/io/trino/sql/query/TestColumnMask.java

```diff
@@ -537,7 +537,24 @@ public void testDeleteWithColumnMasking()
                 USER,
                 new ViewExpression(USER, Optional.empty(), Optional.empty(), "clerk"));
         assertThatThrownBy(() -> assertions.query("DELETE FROM orders"))
-                .hasMessageMatching("\\Qline 1:1: Delete from table with column mask");
+                .hasMessage("line 1:1: Delete from table with column mask");
+    }
+
+    @Test
+    public void testUpdateWithColumnMasking()
+    {
+        accessControl.reset();
+        accessControl.columnMask(
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "clerk",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "clerk"));
+        assertThatThrownBy(() -> assertions.query("UPDATE orders SET clerk = 'X'"))
+                .hasMessage("line 1:1: Updating a table with column masks is not supported");
+        assertThatThrownBy(() -> assertions.query("UPDATE orders SET orderkey = -orderkey"))
+                .hasMessage("line 1:1: Updating a table with column masks is not supported");
```

Comment:
`orderkey` here is not masked, but updates are still disallowed. Should updating only non-masked columns be allowed?